google/data_source_google_service_account_key_test.go

```diff
@@ -0,0 +1,48 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDatasourceGoogleServiceAccountKey_basic(t *testing.T) {
+	t.Parallel()
+
+	resourceName := "data.google_service_account_key.acceptance"
+	account := acctest.RandomWithPrefix("tf-test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccDatasourceGoogleServiceAccountKey(account),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleServiceAccountKeyExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "name"),
```

Comment:
It'd be great to test that they're set to the right values, also.

Reply:
The service account key endpoint returns the name as `projects/PROJECT-ID/serviceAccounts/SA-NAME@PROJECT-ID.iam.gserviceaccount.com/keys/<key_id>` - should I check that it starts with the expected service account?

Reply:
Well, right now I think this just checks that the resource has any attribute as `name` - if you could check that it's the value you expect somehow, that would help.